apps/mosaico/src/Mosaico/Header.purs

```diff
@@ -20,160 +23,213 @@ import React.Basic (JSX)
+
+type Props
+  = { router :: PushStateInterface
+    , categoryStructure :: Array Category
+    , catMap :: Categories
+    , onCategoryClick :: Category -> EventHandler
+    , onLogin :: EventHandler
+    , onProfile :: EventHandler
+    , onStaticPageClick :: String -> EventHandler
+    , user :: Maybe User
+    }
 
-type Props =
-  { router :: PushStateInterface
-  , categoryStructure :: Array Category
-  , catMap :: Categories
-  , onCategoryClick :: Category -> EventHandler
-  , onLogin :: EventHandler
-  , onProfile :: EventHandler
-  , onStaticPageClick :: String -> EventHandler
-  , user :: Maybe User
-  }
-
-render :: Props -> JSX
-render props =
-  DOM.header
-    { className: block
-    , children:
-        [ DOM.div
-            { className: block <> "__left-links"
-            , children:
-                [ DOM.a
-                    { href: "/sida/kontakt"
-                    , onClick: props.onStaticPageClick "kontakt"
-                    , children: [ DOM.text "KONTAKTA OSS" ]
-                    }
-                , DOM.text "|"
-                , DOM.a
-                    { children: [ DOM.text "E-TIDNINGEN" ]
-                    , href: "/epaper"
-                    , onClick: capture_ $ props.router.pushState (write {}) "/epaper"
-                    }
-                ]
-            }
-        , DOM.div
-            { className: block <> "__right-links"
-            , children:
-                [ DOM.ul_
-                    [ DOM.li_
-                        [ DOM.a 
-                            { children: [ DOM.text "KUNDSERVICE" ]
-                            , href: "/sida/kundservice"
-                            , onClick: props.onStaticPageClick "kundservice"
-                            }
-                        ]
-                    , DOM.li_
-                        [ DOM.a
-                            { className: block <> "__prenumerera-link"
-                            , children: [ DOM.text "PRENUMERERA" ]
-                            , href: "https://prenumerera.ksfmedia.fi/#/" <> String.toLower (toString mosaicoPaper)
-                            , target: "_blank"
-                            }
-                        ]
-                    ]
-                ]
-            }
-        , DOM.div
-            { className: block <> "__logo"
-            , onClick: foldMap props.onCategoryClick frontpageCategory
-            }
-        , maybe
-            (DOM.div
-               { children: [ DOM.text "LOGGA IN" ]
-               , onClick: props.onLogin
-               , className: accountClass <> " " <> accountClass <> "--active"
-               , _data: Object.fromFoldable [Tuple "login" "1"]
-               }
-            )
-            (\name ->
-                DOM.a
-                  { className: accountClass
-                  , onClick: props.onProfile
-                  , href: "/konto"
+header :: Props -> JSX
+header = unsafePerformEffect mkHeader
```

Comment:
The fact that creating components is effectful kinda sucks but let's try to avoid this.